README.md

```diff
@@ -4,33 +4,301 @@ sktm - skt manager
+into the `password` field value of the configuration file created above. Note:
+secure your configuration file appropriately when configuring production
+systems.
+
+Test that Jenkins Job Builder can reach and authenticate to Jenkins by running
+`jenkins-jobs list`. It should output something like `INFO:root:Matching jobs:
+0` and complete succesfully.
+
+The Jenkins project description ([example-project.yaml](example-project.yaml))
+used in this documentation will run an HTTP server on port 4040 to let Beaker
+download the tested kernel images. If your Jenkins host is behind NAT (e.g. in
+a VM), make sure to setup port redirection, to have it accessible both
+from the Beaker instance and from inside the host, at the same address, and
+change the URL skt will be publishing with:
+
+    sed -ie 's/`hostname -f`:4040/<HOSTNAME>:<PORT>/' example-project.yaml
+
+Here, `<HOSTNAME>` would be the name of the NAT host Beaker would need to
+access, and `<PORT>` would be the redirected port number. E.g.:
+
+    sed -ie 's/`hostname -f`:4040/pc.example.com:4040/' example-project.yaml
+
+You will need to update the file again, if your NAT host name changes.
+
+Create a Jenkins project for sktm to trigger, from the
+[`example-project.yaml`](example-project.yaml) file, using Jenkins Job
+Builder:
+
+    jenkins-jobs update example-project.yaml
+
+Then go to the created project in Jenkins UI and click "[Build
+Now](http://localhost:8080/job/sktm/build)" to check that it works, and to
+have the pipeline parameters (which sktm requires) registered by Jenkins.
+
+Afterwards, every time you would change your YAML project description you will
+need to re-run `jenkins-jobs update example-project.yaml` and build the
+project manually once, as described above.
+
+### SKTM
+
+Create the `~/.sktmrc` file telling sktm how to access Jenkins:
+
+    cat >~/.sktmrc <<EOF
     [config]
-    jurl = http://jenkins.baseurl.com:8080
-    jlogin = mylogin
-    jpass = mypassword
-    jjname = jenkinsjobname
+    jurl = http://localhost:8080
+    jlogin = sktm
+    jpass = sesame
+    EOF
+
+Usage
+-----
+On the first execution sktm creates a database file (`~/.sktm.db` by default),
+which is then used to track status of tested kernel branches, patches, and
+the execution of Jenkins jobs sktm submits.
+
+### Establishing baseline
+
+Before beginning testing a kernel branch, you need to establish a working
+"baseline" commit, the patches would be applied on:
+
+    ./sktm.py -v --jjname <JENKINS_PROJECT> baseline <GIT_REPO_URL> <GIT_REF>
+
+Here, `<JENKINS_PROJECT>` would be the name of the Jenkins project running
+skt, `<GIT_REPO_URL>` would be a kernel Git repository URL, and `<GIT_REF>`
+would be a published Git reference in that repository from which to start
+locating a stable "baseline" commit, and usually will be a branch name.
+
+E.g., following our setup example above, this command would find a baseline
```

Comment:
All the jobs submitted on Fedora 27 fail with [curl: (16) Error in the HTTP2 framing layer](https://github.com/curl/curl/issues/1618) when downloading kernel config. While it doesn't looks like a problem on our side, I'm posting it here so nobody else is surprised.

Reply:
Whoa, I'll check if I see this on F28 and mention in the text, if it's still there.